lib/code_corps/github/events/installation.ex

```diff
@@ -32,80 +32,77 @@ defmodule CodeCorps.GitHub.Events.Installation do
+    |> Changeset.put_assoc(:user, user)
+    |> Repo.insert()
   end
 
-  defp find_user(%{"id" => github_id}), do: User |> Repo.get_by(github_id: github_id)
-
-  defp find_installation(%{"id" => github_id}), do: GithubAppInstallation |> Repo.get_by(github_id: github_id)
+  defp create_matched_installation(%GithubAppInstallation{} = installation) do
+    installation
+    |> start_installation_processing()
+    |> process_repos()
+    |> stop_installation_processing()
+  end
 
-  defp create_unmatched_user_installation(%{"id" => github_id}) do
-    %GithubAppInstallation{}
-    |> Changeset.change(%{state: "unmatched_user", github_id: github_id, installed: true})
+  defp create_repository(%GithubAppInstallation{} = installation, repo_attributes) do
+    %GithubRepo{}
+    |> Changeset.change(repo_attributes |> GithubRepoAdapter.from_api)
+    |> Changeset.put_assoc(:github_app_installation, installation)
     |> Repo.insert()
   end
 
-  defp create_installation_initiated_on_github(%User{} = user, %{"id" => github_id}) do
+  defp create_unmatched_user_installation(%{"id" => github_id}) do
     %GithubAppInstallation{}
-    |> Changeset.change(%{state: "initiated_on_github", github_id: github_id, installed: true})
-    |> Changeset.put_assoc(:user, user)
+    |> Changeset.change(%{github_id: github_id, installed: true, state: "unmatched_user"})
     |> Repo.insert()
   end
 
-  defp update_installation(%User{} = user, %GithubAppInstallation{} = github_app_installation) do
-    # TODO: user and installation need to relate. What if they don't?
-
-    github_app_installation
-    |> start_app_processing()
-    |> process_repos(user)
-    |> stop_app_processing()
-  end
+  defp find_installation(%{"id" => github_id}), do: GithubAppInstallation |> Repo.get_by(github_id: github_id)
 
-  defp start_app_processing(%GithubAppInstallation{} = github_app_installation) do
-    github_app_installation
-    |> Changeset.change(%{state: "processing", installed: true})
-    |> Repo.update()
-  end
+  defp find_user(%{"id" => github_id}), do: User |> Repo.get_by(github_id: github_id)
 
-  defp process_repos({:ok, %GithubAppInstallation{} = github_app_installation}, %User{} = user) do
+  defp process_repos({:ok, %GithubAppInstallation{} = installation}) do
     # TODO: Consider moving into transaction
     {:ok, repositories} =
-      user
-      |> GitHub.Installation.repositories(github_app_installation)
+      installation
+      |> GitHub.Installation.repositories()
       |> (fn {:ok, repositories} -> repositories end).()
-      |> Enum.map(&create_repository(github_app_installation, &1))
+      |> Enum.map(&create_repository(installation, &1))
       |> Enum.map(fn {:ok, repository} -> repository end)
       |> (fn repositories -> {:ok, repositories} end).()
 
-    {:ok, github_app_installation |> Map.put(:github_repos, repositories)}
+    {:ok, installation |> Map.put(:github_repos, repositories)}
```

Comment:
This was me trying to aggregate the results into something useful, but it isn't strictly necessary. Basically, everything after line 80 is unneeded. I'm fine with leaving it in, but also not sure about it causing confusion.